networkit/cpp/graph/GraphBuilder.cpp

```diff
@@ -366,4 +378,97 @@ count GraphBuilder::numberOfEdges(const Graph &G) {
     }
 }
 
+// Parallel functions
+
+void GraphBuilder::addEdgeParallel(node a, node b) {
```

Comment:
See comment at function `addEdgesInParallel`: If the new concept is faster (also for one thread), then drop the previous `addHalfEdge` function (and others) in favor of the new one(s).  If for the sequential case, nothing is gained, a wrapper/template is needed to distinguish between sequential and parallel insertion.